File: frontend/lib/features/address/domain/entities/address.dart
class Address {
  final String id;
  final String alias;
  final String recipientName;
  final String phone;
  final String street;
  final String number;
  final String? apartment;
  final String city;
  final String state;
  final String postalCode;
  final String country;
  final bool isDefault;
  final String? instructions;
  final DateTime createdAt;
  final DateTime updatedAt;

  Address({
    required this.id,
    required this.alias,
    required this.recipientName,
    required this.phone,
    required this.street,
    required this.number,
    this.apartment,
    required this.city,
    required this.state,
    required this.postalCode,
    required this.country,
    required this.isDefault,
    this.instructions,
    required this.createdAt,
    required this.updatedAt,
  });

  factory Address.fromJson(Map<String, dynamic> json) {
    return Address(
      id: json['_id'],
      alias: json['alias'],
      recipientName: json['recipientName'],
      phone: json['phone'],
      street: json['street'],
      number: json['number'],
      apartment: json['apartment'],
      city: json['city'],
      state: json['state'],
      postalCode: json['postalCode'],
      country: json['country'],
      isDefault: json['isDefault'] ?? false,
      instructions: json['instructions'],
      createdAt: DateTime.parse(json['createdAt']),
      updatedAt: DateTime.parse(json['updatedAt']),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'alias': alias,
      'recipientName': recipientName,
      'phone': phone,
      'street': street,
      'number': number,
      if (apartment != null && apartment!.isNotEmpty) 'apartment': apartment,
      'city': city,
      'state': state,
      'postalCode': postalCode,
      'country': country,
      'isDefault': isDefault,
      if (instructions != null && instructions!.isNotEmpty)
        'instructions': instructions,
    };
  }

  String get fullAddress {
    final parts = [
      '$street $number',
      if (apartment != null && apartment!.isNotEmpty) 'Piso/Apto: $apartment',
      '$city, $state',
      postalCode,
      country,
    ];
    return parts.where((p) => p.isNotEmpty).join(', ');
  }
}

File: backend/src/models/Address.js
import mongoose from 'mongoose';

const addressSchema = new mongoose.Schema({
  user: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User',
    required: true,
    index: true
  },
  alias: {
    type: String,
    required: [true, 'El alias es requerido'],
    trim: true,
    maxlength: [50, 'El alias no puede tener más de 50 caracteres']
  },
  recipientName: {
    type: String,
    required: [true, 'El nombre del destinatario es requerido'],
    trim: true
  },
  phone: {
    type: String,
    required: [true, 'El teléfono es requerido'],
    trim: true
  },
  street: {
    type: String,
    required: [true, 'La calle es requerida'],
    trim: true
  },
  number: {
    type: String,
    required: [true, 'El número es requerido'],
    trim: true
  },
  apartment: {
    type: String,
    trim: true,
    default: ''
  },
  city: {
    type: String,
    required: [true, 'La ciudad es requerida'],
    trim: true
  },
  state: {
    type: String,
    required: [true, 'La provincia/estado es requerida'],
    trim: true
  },
  postalCode: {
    type: String,
    required: [true, 'El código postal es requerido'],
    trim: true
  },
  country: {
    type: String,
    required: [true, 'El país es requerido'],
    default: 'Argentina',
    trim: true
  },
  isDefault: {
    type: Boolean,
    default: false
  },
  instructions: {
    type: String,
    trim: true,
    maxlength: [200, 'Las instrucciones no pueden tener más de 200 caracteres']
  }
}, {
  timestamps: true
});

// Índice compuesto para evitar alias duplicados por usuario
addressSchema.index({ user: 1, alias: 1 }, { unique: true });

// Middleware: asegurar que solo una dirección es default por usuario
addressSchema.pre('save', async function(next) {
  if (this.isDefault) {
    await this.constructor.updateMany(
      { user: this.user, _id: { $ne: this._id } },
      { isDefault: false }
    );
  }
  next();
});

const Address = mongoose.model('Address', addressSchema);
export default Address;